store/tikv/mocktikv/cop_handler_dag.go

```diff
@@ -337,6 +373,98 @@ func flagsToStatementContext(flags uint64) *stmtctx.StatementContext {
 	return sc
 }
 
+type mockClientStream struct{}
+
+func (mockClientStream) Header() (metadata.MD, error) { return nil, nil }
+func (mockClientStream) Trailer() metadata.MD         { return nil }
+func (mockClientStream) CloseSend() error             { return nil }
+func (mockClientStream) Context() goctx.Context       { return nil }
+func (mockClientStream) SendMsg(m interface{}) error  { return nil }
+func (mockClientStream) RecvMsg(m interface{}) error  { return nil }
+
+type mockCoprocessorStreamClient struct {
+	mockClientStream
+	req      *tipb.DAGRequest
+	exec     executor
+	goCtx    goctx.Context
+	finished bool
+}
+
+type streamRegionError struct {
+	mockClientStream
+	*errorpb.Error
+}
+
+func (e *streamRegionError) Recv() (*coprocessor.Response, error) {
+	return &coprocessor.Response{
+		RegionError: e.Error,
+	}, nil
+}
+
+func (mock *mockCoprocessorStreamClient) Recv() (*coprocessor.Response, error) {
+	if mock.finished {
+		return nil, io.EOF
+	}
+
+	var resp coprocessor.Response
+	chunk, finish, err := readBlockFromExecutor(mock.goCtx, mock.exec, mock.req.OutputOffsets)
+	if err != nil {
+		if locked, ok := errors.Cause(err).(*ErrLocked); ok {
+			resp.Locked = &kvrpcpb.LockInfo{
+				Key:         locked.Key,
+				PrimaryLock: locked.Primary,
+				LockVersion: locked.StartTS,
+				LockTtl:     locked.TTL,
+			}
+		} else {
+			resp.OtherError = err.Error()
+		}
+		return &resp, nil
+	}
+	if finish {
+		// Just mark it, need to handle the last chunk.
+		mock.finished = true
+	}
+
+	data, err := chunk.Marshal()
+	if err != nil {
+		resp.OtherError = err.Error()
+		return &resp, nil
+	}
+	streamResponse := tipb.StreamResponse{
+		Error:      toPBError(err),
+		EncodeType: tipb.EncodeType_TypeDefault,
+		Data:       data,
+	}
+	data1, err1 := proto.Marshal(&streamResponse)
+	if err1 != nil {
+		resp.OtherError = err1.Error()
+		return &resp, nil
+	}
+	resp.Data = data1
+	return &resp, nil
+}
+
+func readBlockFromExecutor(goCtx goctx.Context, e executor, rowCols []uint32) (tipb.Chunk, bool, error) {
+	var err error
+	var count int
+	var chunk tipb.Chunk
+	for count = 0; count < rowsPerChunk; count++ {
+		var row [][]byte
+		row, err = e.Next(goCtx)
```

Comment:
s/row, err = e.Next(goCtx)/row, err := e.Next(goCtx)/, we can remove `var err error` and `var row [][]byte`